applications/dashboard/models/class.logmodel.php

```diff
@@ -796,16 +811,32 @@ private function restoreOne($Log, $DeleteLog = true) {
                         Gdn::userModel()->setField($ID, 'Banned', 0);
                     }
 
-                    // Keep track of a discussion ID so that its count can be recalculated.
-                    if ($Log['Operation'] != 'Edit') {
-                        switch ($Log['RecordType']) {
-                            case 'Discussion':
-                                $this->recalcIDs['Discussion'][$ID] = true;
-                                break;
-                            case 'Comment':
-                                $this->recalcIDs['Discussion'][$Log['ParentRecordID']] = true;
-                                break;
-                        }
+                    // Keep track of discussions and categories so that there counts can be recalculated.
+                    switch ($Log['RecordType']) {
+                        case 'Discussion':
+                            $this->recalcIDs['Discussion'][$ID] = true;
+
+                            if (empty($this->recalcIDs['Category'][$Log['CategoryID']])) {
+                                $this->recalcIDs['Category'][$Log['CategoryID']] = [
+                                    'CountDiscussions' => 1,
+                                    'CountComments' => 0,
+                                ];
+                            } else {
+                                $this->recalcIDs['Category'][$Log['CategoryID']]['CountDiscussions']++;
+                            }
+                            break;
+                        case 'Comment':
+                            $this->recalcIDs['Discussion'][$Log['ParentRecordID']] = true;
+
+                            if (empty($this->recalcIDs['Category'][$Log['CategoryID']])) {
```

Comment:
`CategoryID` does not appear to be a valid field for comment log entries. `ParentRecordID` is populated with the discussion's ID on a comment log row. `CategoryID` is populated for discussions.